core/src/main/java/org/apache/accumulo/core/clientImpl/NamespaceOperationsImpl.java

```diff
@@ -158,6 +163,12 @@ public void delete(String namespace) throws AccumuloException, AccumuloSecurityE
 
     try {
       doNamespaceFateOperation(FateOperation.NAMESPACE_DELETE, args, opts, namespace);
+    } catch (AccumuloException ae) {
+      if (ae.getMessage().contains(TABLES_EXISTS_IN_NAMESPACE_INDICATOR)) {
+        throw new NamespaceNotEmptyException(namespaceId.canonical(), namespace, null, ae);
+      } else {
+        throw ae;
+      }
     } catch (NamespaceExistsException e) {
       // should not happen
       throw new AssertionError(e);
```

Comment:
Instead of relying on the special String to detect this situation, you could instead use this existing NamespaceExistsException as a special case with the semantics of "namespace still exists (because it wasn't empty and couldn't be deleted)".